source/Engine.cpp

```diff
@@ -1461,10 +1461,26 @@ void Engine::CalculateStep()
 
 	// Keep track of the flagship to see if it jumps or enters a wormhole this turn.
 	const Ship *flagship = player.Flagship();
+	bool flagshipWasUntargetable = (flagship && !flagship->IsTargetable());
 	bool wasHyperspacing = (flagship && flagship->IsEnteringHyperspace());
-	// Move all the ships.
+	const System *flagshipSystem = (flagship ? flagship->GetSystem() : nullptr);
+	// First, move the player's flagship.
+	MoveShip(player.FlagshipPtr());
```

Comment:
Let's grab the system after moving. I imagine it won't actually change the behavior, but it's more correct.
```suggestion
	// First, move the player's flagship.
	MoveShip(player.FlagshipPtr());
	const System *flagshipSystem = (flagship ? flagship->GetSystem() : nullptr);
```